cdap-api/src/main/java/io/cdap/cdap/api/app/AppStateWriter.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.app;
+
+/**
+ * Save state for the app
+ */
+public interface AppStateWriter {
+
+  /**
+   * Method for saving state for an app with a key
+   * @param namespace Namespace for the app
+   * @param appName Name of the app
+   * @param key Key for the state
+   * @param value value as byte[]
+   * @throws AppStateException if the app is not available or otherwise unable to save state
+   */
+  void saveState(String namespace, String appName, String key, byte[] value) throws AppStateException;
```

Comment:
similar comment to above, I think we only need the key and value

Reply:
Done.